website/project/views/discussions.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+
+import re
+
+from flask import request
+from modularodm import Q
+from modularodm.exceptions import NoResultsFound
+
+from framework.auth.decorators import collect_auth
+from framework.auth.core import User
+
+from website import mails
+from website.project.decorators import (
+    must_be_valid_project,
+    must_have_permission,
+    must_not_be_registration,
+    must_be_contributor
+)
+from website.util.permissions import ADMIN
+from website.models import Node
+
+
+###############################################################################
+# Internal Calls
+###############################################################################
+
+
+@must_be_valid_project
+@must_have_permission(ADMIN)
+@must_not_be_registration
+def enable_discussions(node, **kwargs):
+    node.discussions.enable(save=True)
+
+
+@must_be_valid_project
+@must_have_permission(ADMIN)
+@must_not_be_registration
+def disable_discussions(node, **kwargs):
+    node.discussions.disable(save=True)
+
+
+@must_be_valid_project
+@collect_auth
+@must_be_contributor
+@must_not_be_registration
+def set_subscription(node, auth, **kwargs):
+    subscription = request.json.get('discussionsSub')
+    subscribed = True if subscription == 'subscribed' else False
+    if subscribed:
+        node.discussions.subscribe_member(auth.user._id, save=True)
+    else:
+        node.discussions.unsubscribe_member(auth.user._id, save=True)
+
+
+###############################################################################
+# MailGun Calls
+###############################################################################
+
+
+def route_message(**kwargs):
+    message = request.form
+    target_address = message['To']
+    node_id = re.search(r'[a-z0-9]*@', target_address).group(0)[:-1]
+    sender_email = message['From']
+    # allow for both "{email}" syntax and "{name} <{email}>" syntax
+    if ' ' in sender_email:
+        sender_email = re.search(r'<\S*>$', sender_email).group(0)[1:-1]
+
+    sender = User.find_by_email(sender_email)
```

Comment:
Consider using `framework.auth.core.get_user` instead, which will return only one user.

Also note that the existence of a user is not the same thing as the existence of a *valid* user. Consider checking `User.is_active` to make sure you're not sending on behalf of an unregistered or unconfirmed user.